File: arrays_and_strings/number_of_ways_to_split_array.js
/**
 * @param {number[]} nums
 * @returns {number}
 */
var waysToSplitArray1 = function(nums) {
    let prefix = [nums[0]];

    for (let i = 1; i < nums.length; i++)
    {
        prefix.push(nums[i] + prefix[prefix.length - 1]);
    }

    let ans = 0;
    for (let i = 0; i < nums.length - 1; i++)
    {
        let leftSection = prefix[i];
        let rightSection = prefix[prefix.length - 1] - prefix[i];
        if (leftSection >= rightSection)
        {
            ans++;
        }
    }

    return ans;
}

var waysToSplitArray2 = function(nums) {
    let ans = 0, leftSection = 0, total = 0;

    for (const num of nums)
    {
        total += num;
    }

    for (let i = 0; i < nums.length - 1; i++)
    {
        leftSection += nums[i];
        const rightSection = total - leftSection;
        if (leftSection >= rightSection)
        {
            ans++;
        }
    }

    return ans;
}

File: arrays_and_strings/number_of_ways_to_split_array.java
public int waysToSplitArray1(int[] nums)
{
    long[] prefix = new long[nums.length];
    prefix[0] = nums[0];

    for (int i = 1; i < nums.length; i++)
    {
        prefix[i] = nums[i] + prefix[i - 1];
    }

    int ans = 0;
    for (int i = 0; i < nums.length - 1; i++)
    {
        long leftSection = prefix[i];
        long rightSection = prefix[nums.length - 1] - prefix[i];
        if (leftSection >= rightSection)
        {
            ans++;
        }
    }

    return ans;
}

public int waysToSplitArray2(int[] nums)
{
    int  ans = 0;
    long leftSection = 0;
    long total = 0;

    for (int num : nums)
    {
        total += num;
    }

    for (int i = 0; i < nums.length - 1; i++)
    {
        leftSection += nums[i];
        const long rightSection = total - leftSection;
        if (leftSection >= rightSection)
        {
            ans++;
        }
    }

    return ans;
}